client/scripts/DrawTogether.js

```diff
@@ -1295,6 +1326,262 @@ DrawTogether.prototype.handlePaintSelection = function handlePaintSelection (eve
 	}.bind(this));
 };
 
+// Send null for unchanging value
+DrawTogether.prototype.updateIndividualFavoriteDom = function updateIndividualFavoriteDom(newX, newY, newName, newOwner, element) {
+	if(newX !== null)
+		element.dataset.x = newX;
+	if(newY !== null)
+		element.dataset.y = newY;
+	if(newName !== null)
+		element.dataset.name = newName;
+	if(newOwner !== null)
+		element.dataset.owner = newOwner;
+	
+	var coordinateButton = element.getElementsByClassName("fav-coor-button")[0];
+	var inputRename = element.getElementsByClassName("fav-rename-input")[0];
+	if ( (newName || element.dataset.name) === ""){
+		inputRename.value = "";
+		coordinateButton.innerHTML = (newX || element.dataset.x) + "," + (newY || element.dataset.y);
+		coordinateButton.title = "";
+	}
+	else {
+		inputRename.value = (newName || element.dataset.name);
+		coordinateButton.innerHTML = (newName || element.dataset.name);
+		coordinateButton.title = (newX || element.dataset.x) + "," + (newY || element.dataset.y);
+	}
+	
+	
+};
+
+DrawTogether.prototype.insertOneFavorite = function insertOneFavorite(x, y, name, owner) {
+	var favoriteContainer = this.favoritesContainer.appendChild(document.createElement("div"));
+	favoriteContainer.className = "favorite-container";
+	favoriteContainer.dataset.x = x;
+	favoriteContainer.dataset.y = y;
+	favoriteContainer.dataset.name = name;
+	favoriteContainer.dataset.owner = owner;
+	
+	var favoriteMinusButton = favoriteContainer.appendChild(document.createElement("div"));
+	favoriteMinusButton.className = "fav-button fav-minus-button";
+	favoriteMinusButton.innerHTML = "-";
+	favoriteMinusButton.addEventListener("click", function (e) {
```

Comment:
![](http://i.imgur.com/TwXQ40M.gif)

it would be hard to separate it into functions but im not certain. tell me if it's important and ill focus on it